### PR TITLE
Changed CanCraft to check if OnCraft returns true

### PR DESCRIPTION
**Pull request by TacticalToaster**
I thought that a custom check would be useful for crafting recipes involving custom systems. You could use OnCraft to check for a type of entity near the player for the recipe or, if you're spawning an entity, check to see if there is room in the direction you're facing. PostCraft could be used for things done on a successful craft.

### CODE REVIEW

**Comment by murray-elijah** [comment]
Verified as working.

**Comment by xRJx** [comment]
I like this change, it still maintains the previous sort of functionality while opening up more approaches to development.

There is a new use case to note however: The idea of OnCraft how it was previously used was that it would be called just prior to a successful craft, so for example a programmer could work with a player's items before they're taken away via crafting.

The new use case is that now OnCraft could be called when crafting fails, so the point I'm trying to make is that the expected time that OnCraft was called has changed: it was always called just prior to a successful craft, whereas now the certainty of it being called on a successful craft is no longer there. How this will affect current/future code that utilizes the OnCraft function is not known to me, but I wanted to make a note of it.

Anyways, I think the commit would be fine as it is and it shouldn't cause much -if any- issues regarding people's code that's currently utilizing the Crafting Library.

In conclusion; I think it would be better to leave the OnCraft callback where it is and then add in new callbacks to facilitate what you're trying to do. Here's some of the code for doing that:
```LUA
--[[
	@codebase Shared
	@details Called when checking if an item can be crafted.
	@param {Entity} Player crafting the blueprint.
--]]
function CLASS_TABLE:CanCraft(player) end;

if (blueprintTable.CanCraft) then
	return blueprintTable:CanCraft(player);
end;
```

**Comment by murray-elijah** [comment]
> Thoughts?

Probably for the best. That, or maybe a partial rework of the crafting system.

**Comment by xRJx** [comment]
See #440 for associated pull request.
